### PR TITLE
Bluetooth: tester: stop advertising after timeout in limited advertising

### PR DESCRIPTION
**Pull request by KKopyscinski**
When limited advertising is used, application is exepected to stop
sending advertisings with limited advertising flag after specified
timeout. Previously, advertising was not stopped automatically and
just continued until stop_advertising() was called.

This affects tests GAP/DISC/LIMM/BV-03-C and GAP/DISC/LIMM/BV-04-C

Signed-off-by: Krzysztof Kopyściński <krzysztof.kopyscinski@codecoup.pl>

### CODE REVIEW

**Review by Thalley — CHANGES_REQUESTED** [review]
The code changes seem sane. 

The test case does not specify that the upper tester shall tell the IUT to stop advertising. Is it really okay that this change does it that way? 

I feel that the correct way is that the stack, knowing that the limited discovery mode flag is set, should autonomously stop the advertiser after 180s (or some other value we set for Zephyr, possibly controlled by a Kconfig option) if it hasn't been stopped from the upper layer (e.g. the upper tester). 

Unfortunately we do not have a callback to notify the upper layer for legacy adv (`bt_le_adv_start`), but we do have one for extended: `struct bt_le_ext_adv_cb`. 

In short: I'm unsure whether this way of fixing the test is allowed (i.e. requiring the upper tester to stop the advertising), as we cannot stop other applications from violating this timeout otherwise.

**Comment by KKopyscinski** [comment]
@Thalley I agree that host is better place for this change, as this timeout should exist always, not only when developer implemented it in application. I created a PR with appropriate host patch, that uses value set in Kconfig: https://github.com/zephyrproject-rtos/zephyr/pull/36776
